core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -338,6 +340,9 @@ private[spark] class Executor(
             metricsSystem = env.metricsSystem)
           threwException = false
           res
+        } catch {
+          case _: CommitDeniedException =>
+            throw new TaskKilledException("commit denied")
```

Comment:
I'm not sure we want to just convert it here. I was originally thinking we would fix it up on the driver side because it knows that it explicitly a speculative task and it killed it.  Here we don't know that for sure.

for instance you might have got a commit denied because the stage was aborted due to fetch failure.  That shouldn't show up as killed.